gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -217,12 +217,10 @@ static void gst_tensor_filter_compare_tensors (GstTensorsInfo * info1,
  * @brief Open nn framework.
  */
 #define gst_tensor_filter_open_fw(filter) do { \
-      if (filter->prop.fw_opened == FALSE && filter->fw) { \
+      if (!filter->prop.fw_opened && filter->fw) { \
         if (filter->fw->open != NULL) {\
           if (filter->fw->open (&filter->prop, &filter->privateData) == 0) \
             filter->prop.fw_opened = TRUE; \
```

Comment:
If ```open()``` is NULL while ```fw``` is available, it means that it does NOT need to call open to proceed. Thus, we should set ```fw_opened = TRUE``` if it's NULL as in the previous code.

Reply:
@myungjoo the two lines you highlighted have the same logic, I just replaced `filter->prop.fw_opened == FALSE` with `!filter->prop.fw_opened`

Reply:
There is NOT in there

Reply:
isn't fw_opened a boolean ?

Reply:
> 
> 
> isn't `fw_opened` a boolean ?

Its semantics is a boolean. Its machine representation is int.

Reply:
> 
> 
> There is a NOT in there

The "then" part of Line 220 is ok.
The problem is the "else" part of Line 221 (not 220): when we do not have "open" callback not defined.

Reply:
With the `else`, then `fw_open` is always TRUE. I don't think this is what we want here.

Reply:
The code checks `fw_open` to see if it is FALSE, when `open` fails. But, with the else, fw_open is always TRUE.

Reply:
You need some way of telling that open failed.

Reply:
Ahhhh, sorry, now I understand! I will correct this.